### PR TITLE
OKX: Fix GetAvailableTransferChains bug

### PR DESCRIPTION
**Pull request by shazbert**
# PR Description

Noticed issue when testing #1174. This PR does not return chain when withdraw/deposit chain functionality is not enabled for currency.

```
[ERROR] | LOG | 01/05/2023 10:01:56 | Okx failed to get cryptocurrency deposit address for USDT [chain USDT-zkSync Lite]. Err: specified chain USDT-zkSync Lite not found
```

Fixes # (issue)

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [ ] go test ./... -race
- [ ] golangci-lint run
- [ ] Test X

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: exchanges/okx/okx_wrapper.go [hunk 1]
```diff
@@ -1453,7 +1453,9 @@ func (ok *Okx) GetAvailableTransferChains(ctx context.Context, cryptocurrency cu
 	}
 	chains := make([]string, 0, len(currencyChains))
 	for x := range currencyChains {
-		if !cryptocurrency.IsEmpty() && !strings.EqualFold(cryptocurrency.String(), currencyChains[x].Currency) {
+		if !cryptocurrency.IsEmpty() &&
+			!strings.EqualFold(cryptocurrency.String(), currencyChains[x].Currency) ||
+			(!currencyChains[x].CanDeposit && !currencyChains[x].CanWithdraw) {
```
Comment (gloriousCode) [inline exchanges/okx/okx_wrapper.go:1458]:
```suggestion
		if (!cryptocurrency.IsEmpty() && !strings.EqualFold(cryptocurrency.String(), currencyChains[x].Currency)) ||
			(!currencyChains[x].CanDeposit && !currencyChains[x].CanWithdraw) {
```

**Review by gloriousCode — CHANGES_REQUESTED** [review]
OK excuse me, this works. One highly important change

File: exchanges/okx/okx_wrapper.go [hunk 2]
```diff
@@ -1453,7 +1453,8 @@ func (ok *Okx) GetAvailableTransferChains(ctx context.Context, cryptocurrency cu
 	}
 	chains := make([]string, 0, len(currencyChains))
 	for x := range currencyChains {
-		if !cryptocurrency.IsEmpty() && !strings.EqualFold(cryptocurrency.String(), currencyChains[x].Currency) {
+		if (!cryptocurrency.IsEmpty() && !strings.EqualFold(cryptocurrency.String(), currencyChains[x].Currency)) ||
+			(!currencyChains[x].CanDeposit && !currencyChains[x].CanWithdraw) {
```
Comment (gloriousCode) [inline exchanges/okx/okx_wrapper.go:1457]: `[ERROR] | LOG | 01/05/2023 13:37:54 | Okx failed to get cryptocurrency deposit address for BTC [chain BTC-Lightning]. Err: specified chain BTC-Lightning not found`

```suggestion
		if (!cryptocurrency.IsEmpty() && !strings.EqualFold(cryptocurrency.String(), currencyChains[x].Currency)) ||
			(!currencyChains[x].CanDeposit && !currencyChains[x].CanWithdraw) ||
			!currencyChains[x].MainNet {
```

**Comment by thrasher-** [comment]
nACK [4345f57](https://github.com/thrasher-corp/gocryptotrader/pull/1177/commits/4345f57ce0e984f562ec576aeebc09407a1d2544), this defeats the purpose of getting the "Available transfer chains" for a specified currency. Using something like USDT will now return an empty result. Instead, if a lightning chain is detected, it will need to use this endpoint: https://www.okx.com/docs-v5/en/#rest-api-funding-lightning-deposits which thankfully only occurs for BTC.

**Review by thrasher- — APPROVED** [review]
tACK, thanks for addressing the nit!

**Review by gloriousCode — CHANGES_REQUESTED** [review]
There is still technically an issue here:
`[ERROR] | LOG | 03/05/2023 07:48:05 | Okx failed to get cryptocurrency deposit address for BETH. Err: deposit address with the specified currency code and chain not found`
It does have a chain, but it does not allow deposits and withdrawals. Would require more extensive tweaking to not have an error displayed for this specific usecase which I don't think this PR needs to address.

So just the unmarshalling fix for me please.

**Review by gloriousCode — APPROVED** [review]
tACK! Thank you shazbert cow 🐄

**Pull request merged**